### ISSUE TITLE
SAT>IP Client mode bugs

### ISSUE THREAD

**Issue by lars18th** [catalinii/minisatip#560]
Hi @catalinii ,

I see this error in the log using the SAT>IP client mode:
```
...
PLAY rtsp://192.168.1.50:554/stream=108?src=1&freq=10744&msys=dvbs&mtype=qpsk&pol=h&sr=22000&fec=56&ro=0.35&plts=off&plsc=0&pids=none&pids=401,402,0,400 RTSP/1.0
CSeq: 9
Session: B521F14D
...
```
This is when you're starting to play a channel. Note the `&pids=none&pids=401,402,0,400` !!!
After that in some cases a new `addpids=` command is sended with the same pid list, or a DESCRIBE is used. But in any case the **double pids** is a bug.

Please, can you check it?
Thank you!

**Comment by lars18th** [comment]
Hi @catalinii ,

I found another trouble with the Client Mode:

- Sometimes my SAT>IP server enters in an unkown state. After some debugging I found that: When a client (a client connected to minisatip) stops to play a channel then you send this RSTP command `PLAY rtsp://192.168.1.70:554/stream=1?pids=none RTSP/1.0` to the server. And after a time, when the device is closed then you send the `TEARDOWN`. However, this message it's sended 30 seconds after. So this generates a problem in some cases.

I suggest to add the one tunning option to force the TEARDOWN message when `pids=none`.
You can do it, please?
Thank you!

**Comment by lars18th** [comment]
Hi,

Doesn't anyone else have this problem too?
I use the SAT>IP _client_ functionality and I see this problem **only** when using directly the `satipc` funcionality. Using the `vtuner` for SAT>IP I don't see this error. However, I prefer to overcome the use of `vtuner`, as the `vtuner` has a limited number of pid-list.

Please, comment your troubles, as the `minisatip` is the only SAT>IP client software that (sometimes) blocks my SAT>IP server.

**Comment by lars18th** [comment]
Hi @catalinii ,

When this problem appears the minisatip as client and the server enters in this state:
```
PLAY rtsp://192.168.1.41:554/stream=3?src=1&freq=10744&msys=dvbs&mtype=&pol=h&sr=22000&plsc=0&pids=none RTSP/1.0
CSeq: 50
Session: 8D0FD734
[...]
RTSP/1.0 200 OK
CSeq: 50
Date: Wed, Mar 27 2019 11:48:53 GMT
Session: 8D0FD734
RTP-Info: url=rtsp://192.168.1.41/stream=3

[...]

DESCRIBE rtsp://192.168.1.41:554/stream=3 RTSP/1.0
CSeq: 51
Session: 8D0FD734
Accept: application/sdp
[...]
[27/03 12:49:28.059 AD1]: satipc_reply (sock 9) handle 8, adapter 1:
RTSP/1.0 200 OK
CSeq: 51
Date: Wed, Mar 27 2019 11:48:53 GMT
Content-Base: rtsp://192.168.1.41/
Content-Type: application/sdp
Content-Length: 238

v=0
o=- 1553684797966000 1 IN IP4 192.168.1.41
s=SatIPServer:1 1
t=0 0
m=video 0 RTP/AVP 33
c=IN IP4 0.0.0.0
a=control:stream=3
a=fmtp:33 ver=1.0;src=1;tuner=1,240,1,15,10744,h,dvbs,qpsk,off,0.00,22000,0;pids=none
a=sendonly
```

So, as you can see the problem is related to this:
- When for some reason (I feel because a bug) the minisatip enters in a `pids=none` state, then it never sends the `TEARDOWN` command to the SAT>IP server. And this generates the problem with this particular server, so if it doesn't close the session then it can't continue.

I suggest to put some "client timeout" to close any SAT>IP client session when `pids=all` continue for a time. In fact, it doesn't have sense to maintain open & live (sending DESCRIBE) if the streaming is none (aka pids=none).

I hope you agree.
Thank you!

**Comment by lars18th** [comment]
Hi @catalinii ,

More info!
I isolated the issue to this case:
- Using SAT>IP client mode, you open one HTTP connection to minisatip. And you get the stream. Then close the streaming. Wait a small interval of time (like 5 seconds), then the SAT>IP client session continues OPEN but with pids=none. Then restart the same streaming... the minisatip **never** request to stream the new pids.

See this log:
```
[27/03 13:44:24.829 main]: satipc: set_pid for adapter 1, pid 0, err 0
[27/03 13:44:24.829 main]: satipc: set_pid for adapter 1, pid 17, err 0
[27/03 13:44:24.829 main]: satipc: set_pid for adapter 1, pid 18, err 0
[27/03 13:44:24.829 main]: satipc: set_pid for adapter 1, pid 1023, err 0
[27/03 13:44:24.829 main]: satipc: set_pid for adapter 1, pid 178, err 0
[27/03 13:44:24.829 main]: satipc: set_pid for adapter 1, pid 176, err 0
[27/03 13:44:24.829 main]: satipc: set_pid for adapter 1, pid 175, err 0
[27/03 13:44:24.829 main]: satipc: set_pid for adapter 1, pid 41, err 0
[27/03 13:44:24.829 main]: satipc: commit for adapter 1 pids to add 8, pids to delete 0, expect_reply 1, force_commit 0 want_tune 1 do_tune 1, force_pids 0
[27/03 13:44:24.829 main]: reply -> 7 (192.168.1.38:57203) CL:0 [sock_id 7]:
[27/03 13:44:24.881 AD1]: satipc_reply (sock 9) handle 8, adapter 1:
RTSP/1.0 200 OK
CSeq: 43
Date: Wed, Mar 27 2019 12:43:38 GMT
Session: 9D47F4D4
RTP-Info: url=rtsp://192.168.1.41/stream=491


[27/03 13:44:24.881 AD1]: wp 0 qp 0, expect_reply 1, want_tune 1, force_commit 0, want commit 1
[27/03 13:44:24.881 AD1]: satipc: commit for adapter 1 pids to add 8, pids to delete 0, expect_reply 0, force_commit 0 want_tune 1 do_tune 0, force_pids 0
[27/03 13:44:24.881 AD1]: satipc_http_request (ad 1): sending to sock 9:
PLAY rtsp://192.168.1.41:554/stream=491?src=1&freq=10744&msys=dvbs&mtype=&pol=h&sr=22000&plsc=0&pids=none RTSP/1.0
CSeq: 44
Session: 9D47F4D4


[27/03 13:44:24.882 AD1]: satipc_reply (sock 9) handle 8, adapter 1:
RTSP/1.0 200 OK
CSeq: 44
Date: Wed, Mar 27 2019 12:43:38 GMT
Session: 9D47F4D4
RTP-Info: url=rtsp://192.168.1.41/stream=491


```

So the bug is this:
- When the session is closed, but not the adapter, then any new PLAY commands doesn't include the new pids to add.
And the solution is "force_pids 1" instead of the current "force_pids 0" in this case.

Please, can you review the code?

**Comment by lars18th** [comment]
Hi @lars18th ,

I fixed the problem! :v: 
Please merge the PR #571 

Regards.

**Comment by 9000h** [comment]
for me the url looks a bit strange
plsc=0 and mtype=
how did the original request url look like?

**Comment by lars18th** [comment]
Hi @9000h ,

:open_mouth: 
Yes, the URL is edited... however, these data isn't relevant. The problem is fixed now! Review my PR #571 . It explains the reason.

Regards. :vulcan_salute:

**Comment by lars18th** [comment]
Merged https://github.com/catalinii/minisatip/pull/571
So I close it!
